examples/pennylane_run_quantum_natural_gradient.py

```diff
@@ -0,0 +1,356 @@
+to the parametrization.
+
+In classical neural networks, the above process is known as
+*natural gradient descent*, and was first introduced by
+`Shun-Ichi Amari (1998) <https://www.mitpressjournals.org/doi/abs/10.1162/089976698300017746>`__.
+The standard gradient descent is modified as follows:
+
+.. math:: \theta_{t+1} = \theta_t - \eta F^{-1}\nabla \mathcal{L}(\theta),
+
+where :math:`F` is the `Fisher information matrix <https://en.wikipedia.org/wiki/Fisher_information#Matrix_form>`__.
+The Fisher information matrix acts as a metric tensor, transforming the
+steepest descent in the euclidean parameter space to the steepest descent in the
+distribution space.
+
+The quantum analogue
+^^^^^^^^^^^^^^^^^^^^
+
+In a similar vein, it has been shown that the standard euclidean geometry
+is sub-optimal for optimization of quantum variational algorithms.
+The space of quantum states instead possesses a unique invariant metric
+tensor known as the Fubini-Study metric tensor :math:`g_{ij}`, which can be used to
+construct a quantum analogue to natural gradient descent:
+
+.. math:: \theta_{t+1} = \theta_t - \eta g^{+}(\theta_t)\nabla \mathcal{L}(\theta)
+
+where :math:`g^{+}` refers to the pseudo-inverse.
+
+.. note::
+
+    It can be shown that the Fubini-Study metric tensor reduces
+    to the Fisher information matrix in the classical limit.
+
+    Furthermore, in the limit where :math:`\eta\rightarrow 0`,
+    the dynamics of the system are equivalent to imaginary-time
+    evolution within the variational subspace, as proposed in
+    `McArdle et al. (2018) <https://arxiv.org/abs/1804.03023>`__.
+"""
+
+##############################################################################
+# Block-diagonal metric tensor
+# ----------------------------
+#
+# The block-diagonal approximation to the Fubini-Study metric tensor
+# of a variational quantum circuit can be easily evaluated on quantum hardware.
+#
+# Consider a quantum node represented by the variational quantum circuit
+#
+# .. math::
+#
+#     U(\mathbf{\theta}) = W(\theta_{i+1}, \dots, \theta_{N})X(\theta_{i})
+#     V(\theta_1, \dots, \theta_{i-1}),
+#
+# where all parametrized gates can be written of the form :math:`X(\theta_{i}) = e^{i\theta_i K_i}`.
+# That is, the gate :math:`K_i` is the *generator* of the parametrized operation :math:`X(\theta_i)`
+# corresponding to the :math:`i`-th parameter.
+#
+# For each parametric layer :math:`\ell` in the variational quantum circuit
+# containing :math:`n` parameters, the :math:`n\times n` block-diagonal submatrix
+# of the Fubini-Study tensor :math:`g_{ij}^{(\ell)}` is calculated by:
+#
+# .. math::
+#
+#     g_{ij}^{(\ell)} = \langle \psi_\ell | K_i K_j | \psi_\ell \rangle
+#     - \langle \psi_\ell | K_i | \psi_\ell\rangle
+#     \langle \psi_\ell |K_j | \psi_\ell\rangle
+#
+# where :math:`|\psi_\ell\rangle =  V(\theta_1, \dots, \theta_{i-1})|0\rangle`
+# (that is, :math:`|\psi_\ell\rangle` is the quantum state prior to the application

```

Comment:
Prior? Looks like after? Or do I understand this wrong?

Reply:
Should be prior! Any ideas how to make this more clear, without making the notation too complicated? The one used in the paper would take too much space to explain...

Reply:
See above, so X is the layer?

Reply:
Rewritten the equations and accompanying text to make it more clear :slightly_smiling_face: